tendrl/bridge_common/JobValidator/api.py

```diff
@@ -0,0 +1,312 @@
+                        + "%s. Expected value type is: %s" % (
+                            iParm, expectedType))
+        return(True, "")
+
+    def getFlowParms(self, flowName):
+        """This function will return mandatory and
+
+        optional parameters of the given flow from the loaded yaml
+        """
+        return (self.yamlObj.get("flows", {}).get(
+            flowName, {}).get("inputs", {}).get("mandatory"),
+            self.yamlObj.get("flows", {}).get(
+                flowName, {}).get("inputs", {}).get("optional"))
+
+    def getAtomParms(self, atom):
+        """This function will return mandatory and
+
+        optional parameters of the given atom from the loaded yaml
+        """
+        objType, con, atomName = atom.split(".")
+        return (self.yamlObj.get("object_details", {}).get(
+            objType, {}).get(con, {}).get(atomName, {}).get(
+                "inputs", {}).get("mandatory"),
+            self.yamlObj.get("object_details", {}).get(
+                objType, {}).get(con, {}).get(atomName, {}).get(
+                    "inputs", {}).get("optional"))
+
+    def _checkCustomType(self, customType, inputParm, inputVal):
+        def _check(inputVal, cType=customType):
+            # check whether the custom defined type is decleared!
+            typeDef = self.yamlObj.get('object_details').get(cType.lower())
+            if not typeDef:
+                return (False, "yaml custom type: '%s' details not found" % (
+                    cType))
+
+            # check whether the item(s) type is valid
+            for item in inputVal:
+                itemType = typeDef['attrs'].get(inputParm, {}).get('type')
+                if not PRIMITIVE_TYPES[itemType](item):
+                    return (False, "Invalid parameter type: "
+                            + "%s. Expected value type is: %s" % (
+                                item, itemType))
+            return (True, "")
+
+        # customType is an array
+        if customType.endswith('[]'):
+            # check whether the given input is an array
+            if not PRIMITIVE_TYPES['List'](inputVal):
+                return (False, "Invalid input type: %s. " % (
+                    inputVal) + " Expected value type is an array")
+            return _check(inputVal, customType[:-2])
+        else:
+            # Its a non-array custom type
+            return _check([inputVal])
+
+    def validateApiV1(self, apiJob):
+        if not self.yamlObj:
+            return (False, "Validating schema not loaded!")
+
+        if "cluster_id" not in apiJob:
+            return (False, "Cluster id not found in the api job")
+        if "flow" not in apiJob:
+            return (False, "flow not found in the api job")
+        if "parameters" not in apiJob:
+            return (False, "parameters not found in the api job")
+
+        # Checking flow details!
+        status = self.checkFlow(apiJob["flow"])
```

Comment:
Need to check for the performance issues here. Its not really required to validate all the flows everytime a API requests lands in